### ISSUE TITLE
Excuse me, luasocket is limited the version of the kernel ?  Timeout does not take effect

### ISSUE THREAD

**Issue by pxyove** [diegonehab/luasocket#349]
Excuse me，I'm currently developing with Luasocket，but I'm having some problems. I hope to get your help. Thank you very much.

I have two devices, one normal, one abnormal, In usocket.c->socket_connect()->connect() errno is inconsistent. Both devices are in nonblocking mode.  Connect RET =-1,errno=150(Operation in Progress) for normal devices, and connect RET =-1,errno=0(Success) lead to timeout no working and always blocking for abnormal devices. However, the two devices have different kernels.  Kernel_version =4.4.198 for a normal device, kernel_version=4.14.172 for an abnormal device.

**Comment by alerque** [comment]
No, the version of the kernel by itself shouldn't be a problem (I use it on older and newer kernels). This sounds like something else in your system config and networking stacks on those systems are different.

**Comment by pxyove** [comment]
Thank you for your help, I will share the reason here when I find it.

**Comment by pxyove** [comment]
Thank you for everyone, I have found out the cause of the problem(system-called connect/recv ret=-1,errno=0 lead to timeout no working and always blocking for abnormal devices), It's caused by compile options error for LINUXTHREADS_OLD=y in uClibc-0.9.33.2.so .config, should select UCLIBC_HAS_THREADS_NATIVE=y.

**Comment by alerque** [comment]
Thanks for reporting back on what you found. It sounds like those are uclibc compile isssue, not relevant to this library so I'll close this now.
